ably.ts.d

```diff
@@ -195,72 +205,90 @@ interface LogInfo {
   handler?: (...args) => void;
 }
 
+interface ChannelEvent {
+  state: ChannelState;
+}
 
-// Common Listeners
-type ListenerFn = (...args: Array<any>) => void;
-type PaginatedResultCallback = (Error: ErrorInfo, results: PaginatedResult ) => void;
-type standardCallback = (Error: ErrorInfo, results: any) => void;
-type messageCallback = (Msg: any) => void;
-type errorCallback = (Error: ErrorInfo) => void;
+interface ChannelStateChange {
+  current: ChannelState;
+  previous: ChannelState;
+  reason?: ErrorInfo;
+  resumed: boolean;
+}
 
+interface ConnectionStateChange {
+  current: ConnectionState;
+  previous: ConnectionState;
+  reason?: ErrorInfo;
+  retryIn?: number;
+}
+
+// Common Listeners
+type PaginatedResultCallback<T> = (error: ErrorInfo, results: PaginatedResult<T> ) => void;
+type standardCallback = (error: ErrorInfo, results: any) => void;
+type messageCallback = (message: Message) => void;
```

Comment:
We either need a presenceCallback or we could make this a generic such as `messageCallback<T> = (message: <T>)`, see below for why